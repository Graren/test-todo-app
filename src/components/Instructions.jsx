import React, { Fragment } from "react";

const Instructions = ({ todo }) => {
  return (
    <div className={"instructions-container"}>
      <p className={"instructions-row"}>
        Click on a Todo to select them or create new todos by writting on the
        input field and pressing confirm. Selecting a todo allows you to update
        them and use keyboard shortcuts
      </p>

      {todo && (
        <Fragment>
          <p className={"instructions-row"}>
            CTRL / Command key + D: Delete TODO
          </p>
          <p className={"instructions-row"}>
            CTRL / Command key + T: Toggle Completion
          </p>
          <p className={"instructions-row"}>
            CTRL / Command key + S: Confirm changes to a TODO (only when
            updating)
          </p>
          <p className={"instructions-row"}>
            CTRL / Command key + Arrow up / Arrow Down: Change priority up /
            down
          </p>
        </Fragment>
      )}
    </div>
  );
};

export default Instructions;
