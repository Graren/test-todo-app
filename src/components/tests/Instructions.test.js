/**
 * @jest-environment jsdom
 */
import React from "react";
import { render } from "@testing-library/react";
import Instructions from "../Instructions.jsx";

describe("link tests", function () {
  test("Instructions renders", () => {
    const component = render(<Instructions />);
    expect(component).toMatchSnapshot();
  });

  test("Instruction shows only one paragraph when no todo is selected", async () => {
    const component = render(<Instructions />);
    let paragraph2 = null
    try {
       paragraph2 = component.getByTestId("test-instructions-delete");
       throw new Error("Component was rendered")
    // eslint-disable-next-line no-empty
    } catch(e) {}

    expect(paragraph2).toBeFalsy();
  });

  test("Instruction shows both paragraphs when a todo is selected", async () => {
    const component = render(<Instructions todo={{}} />);

    const paragraph2 = await component.findByTestId("test-instructions-delete");

    expect(paragraph2).toBeTruthy();
  });
});
