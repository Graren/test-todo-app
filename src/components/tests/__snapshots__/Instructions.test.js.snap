// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link tests Instructions renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="instructions-container"
      >
        <p
          class="instructions-row"
        >
          Click on a Todo to select them or create new todos by writting on the input field and pressing confirm. Selecting a todo allows you to update them and use keyboard shortcuts
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="instructions-container"
    >
      <p
        class="instructions-row"
      >
        Click on a Todo to select them or create new todos by writting on the input field and pressing confirm. Selecting a todo allows you to update them and use keyboard shortcuts
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
